processing/src/main/java/org/apache/druid/query/expression/IPv4AddressExprUtils.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.expression;
+
+import com.google.common.net.InetAddresses;
+
+import javax.annotation.Nullable;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.util.regex.Pattern;
+
+class IPv4AddressExprUtils
+{
+  private static final Pattern IPV4_PATTERN = Pattern.compile("(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})");
+
+  /**
+   * @return True if argument cannot be represented by an unsigned integer (4 bytes), else false
+   */
+  static boolean overflowsUnsignedInt(long value)
+  {
+    return value < 0L || 0xff_ff_ff_ffL < value;
+  }
+
+  /**
+   * @return True if argument is a valid IPv4 address dotted-decimal string
+   */
+  static boolean isValidAddress(@Nullable String string)
+  {
+    return string != null && IPV4_PATTERN.matcher(string).matches();
```

Comment:
This'll accept some invalid IP-like strings, such as `300.300.300.300`. It seems like it's mostly being used to avoid exception-throwing, so how about changing the name to `isPossiblyValidAddress` and emphasizing the might-be-false-positive nature of the check?

Or alternatively, making the check exact and keeping the name the same.

Reply:
Ah! I'll change the regex to be tighter and add more test cases.

Reply:
Thanks!